test/test/cpu/CpuTests.java

```diff
@@ -29,7 +29,24 @@ public void ctimerTotal(TestProcess p) throws Exception {
 
     @Test(mainClass = CpuBurner.class)
     public void itimerTotal(TestProcess p) throws Exception {
-        Output out = p.profile("-d 2 -e itimer -i 100ms --total -o collapsed");
-        assertCloseTo(out.total(), 2_000_000_000, "itimer total should match profiling duration");
+        Output out;
+        long cpuTime;
+        long wallTime;
+        try (CpuTimeService timeService = new CpuTimeService(p.pid())) {
+            wallTime = System.currentTimeMillis();
+            cpuTime = timeService.getProcessCpuTimeNanos();
+            out = p.profile("-d 2 -e itimer -i 100ms --total -o collapsed");
+            cpuTime = timeService.getProcessCpuTimeNanos() - cpuTime;
+            wallTime = (System.currentTimeMillis() - wallTime) * 1_000_000;
+        }
+        double ratio = (double)cpuTime / wallTime;
+        if (ratio < 0.5) {
+            System.err.println("Warning: CPU time / wall time ratio is too low: " + ratio);
+            System.err.println("Your system has too high load to run the itimer test properly");
+            System.err.println("Therefore the test is skipped");
```

Comment:
This is not really maintainable, sorry. A simple 2-line test turned into something convoluted.
The purpose of unit tests is to verify basic assumptions (not just to "pass"). But it's no longer clear for a casual reader what this test verifies exactly and what it has to do with the cpu/wall ratio.

Performance related tests may have certain assumptions about environment they run in - this is normal. If assumptions do not hold true in your environment - why don't you just skip this test? There is `SKIP` make variable for this purpose. After all, you already skip this test under load. It's not strictly required for all tests to run everywhere. For example, we already skip PMU tests on systems where there is no access to hardware performance counters.